modules/bowtie2/align/main.nf

```diff
@@ -1,81 +1,73 @@
-    tuple val(meta), path('*fastq.gz'), emit: fastq, optional:true
+    tuple val(meta), path("*.bam")    , emit: bam
+    tuple val(meta), path("*.log")    , emit: log
+    tuple val(meta), path("*fastq.gz"), emit: fastq, optional:true
     path  "versions.yml"              , emit: versions
 
     when:
     task.ext.when == null || task.ext.when
 
     script:
-    def args = task.ext.args ?: ''
-    def args2 = task.ext.args2 ?: ''
+    def args = task.ext.args ?: ""
+    def args2 = task.ext.args2 ?: ""
     def prefix = task.ext.prefix ?: "${meta.id}"
-    if (meta.single_end) {
-        def unaligned = save_unaligned ? "--un-gz ${prefix}.unmapped.fastq.gz" : ''
-        """
-        INDEX=`find -L ./ -name "*.rev.1.bt2" | sed 's/.rev.1.bt2//'`
-        [ -z "\$INDEX" ] && INDEX=`find -L ./ -name "*.rev.1.bt2l" | sed 's/.rev.1.bt2l//'`
-        [ -z "\$INDEX" ] && echo "BT2 index files not found" 1>&2 && exit 1
-        bowtie2 \\
-            -x \$INDEX \\
-            -U $reads \\
-            --threads $task.cpus \\
-            $unaligned \\
-            $args \\
-            2> ${prefix}.bowtie2.log \\
-            | samtools view -@ $task.cpus $args2 -bhS -o ${prefix}.bam -
 
-        cat <<-END_VERSIONS > versions.yml
-        "${task.process}":
-            bowtie2: \$(echo \$(bowtie2 --version 2>&1) | sed 's/^.*bowtie2-align-s version //; s/ .*\$//')
-            samtools: \$(echo \$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\$//')
-            pigz: \$( pigz --version 2>&1 | sed 's/pigz //g' )
-        END_VERSIONS
-        """
+    def unaligned = ""
+    def reads_args = ""
+
+    if (meta.single_end) {
+        unaligned = save_unaligned ? "--un-gz ${prefix}.unmapped.fastq.gz" : ""
+        reads_args = "-U ${reads}"
     } else {
-        def unaligned = save_unaligned ? "--un-conc-gz ${prefix}.unmapped.fastq.gz" : ''
-        """
-        INDEX=`find -L ./ -name "*.rev.1.bt2" | sed 's/.rev.1.bt2//'`
-        [ -z "\$INDEX" ] && INDEX=`find -L ./ -name "*.rev.1.bt2l" | sed 's/.rev.1.bt2l//'`
-        [ -z "\$INDEX" ] && echo "BT2 index files not found" 1>&2 && exit 1
-        bowtie2 \\
-            -x \$INDEX \\
-            -1 ${reads[0]} \\
-            -2 ${reads[1]} \\
-            --threads $task.cpus \\
-            $unaligned \\
-            $args \\
-            2> ${prefix}.bowtie2.log \\
-            | samtools view -@ $task.cpus $args2 -bhS -o ${prefix}.bam -
+        unaligned = save_unaligned ? "--un-conc-gz ${prefix}.unmapped.fastq.gz" : ""
+        reads_args = "-1 ${reads[0]} -2 ${reads[1]}"
+    }
 
-        if [ -f ${prefix}.unmapped.fastq.1.gz ]; then
-            mv ${prefix}.unmapped.fastq.1.gz ${prefix}.unmapped_1.fastq.gz
-        fi
-        if [ -f ${prefix}.unmapped.fastq.2.gz ]; then
-            mv ${prefix}.unmapped.fastq.2.gz ${prefix}.unmapped_2.fastq.gz
-        fi
+    def samtools = "samtools view -@ $task.cpus --fast --with-header ${args2} > ${prefix}.bam"
```

Comment:
```suggestion
    def samtools_command = "samtools view -@ $task.cpus --fast --with-header ${args2} > ${prefix}.bam"
```

Reply:
This command has been changed based on what we had before in terms of options?

```
samtools view -@ $task.cpus $args2 -bhS -o ${prefix}.bam -
```

Reply:
the command is essentially the same, but uses long form args and some faster compression to bam